### DISCUSSION TITLE
Что не так с DateInput и DateRangeInput на мобильных?

### DISCUSSION THREAD

**Discussion by Kritolan** [VKCOM/VKUI#7617] [Q&A]
В документации сказано, что эти компоненты предназначены для desktop, но не сказано почему и какие подводные камни могут быть при использовании на мобильных устройствах.

Что в таком случае использовать для выбора даты/времени? Вариант с селектами выглядит несколько абсурдно т.к. для выбора временного диапазона потребуется их 10 штук. Только кастомные компоненты?

**Comment by SevereCloud** [comment]
Проблема в основном в выезжающем календаре, он не предназначен для телефонов

---

Самый простой вариант это использовать [Input ](https://vkcom.github.io/VKUI/#!/Input) с `type: 'date' | 'time' | 'datetime-local' | ...`

```tsx
<Input type="date" />;
<Input type="time" />;
<Input type="datetime-local" />;
```

![image](https://github.com/user-attachments/assets/76845bf7-22b8-422f-96dc-85ca7d543afb)

Но нужно понимать что идеального поля ввода времени не существует, нужно учитывать контекст в котором оно будет использоваться

- см. #7070

**Reply by Kritolan** [reply]
Не предназначена конкретно эта реализация компонента? Или в целом?

Про `Input` совершенно не очевидно т.к. в документации нет ни слова про поддержку других типов, кроме "text".

**Reply by SevereCloud** [reply]
> Не предназначена конкретно эта реализация компонента? Или в целом?

DateInput, DateRangeInput и Calendar не предназначены для мобильных устройств.

> Про Input совершенно не очевидно т.к. в документации нет ни слова про поддержку других типов, кроме "text".

Подумаем о том, чтобы возможно добавить это в документацию

**Reply by Kritolan** [reply]
> DateInput, DateRangeInput и Calendar не предназначены для мобильных устройств.

А почему? И будет ли эта поддержка в будущем?

Использовать нативные элементы здорово, но только пока весь остальной интерфейс нативен, иначе это сразу выбивается из всего интерфейса и наоборот сбивает пользователя с толку.

**Reply by SevereCloud** [reply]
В будущем возможно поддержка будет. Сейчас для кроссплатформенности лучше использовать нативные. Либо сделать [свой инпут](https://github.com/VKCOM/VKUI/discussions/7070) с нужным форматом даты для более лучшего пользовательского опыта в зависимости от контекста

**Reply by Kritolan** [reply]
Но на главный вопрос в заголовке - в чем именно заключаются проблемы на телефонах так никто не ответил.

**Reply by SevereCloud** [reply]
> Но на главный вопрос в заголовке - в чем именно заключаются проблемы на телефонах так никто не ответил.

> Проблема в основном в выезжающем календаре, он не предназначен для телефонов

Календарь, не предназначен для телефонов, он не вмещается в телефоны

**Reply by Kritolan** [reply]
А что именно под этим подразумевается? Вот пример как это выглядит сейчас на живом телефоне - вроде как все вполне влезает и вполне удобно тыкается. Да, возможно через popup было бы несколько удобнее - но и текущий вид значительно удобнее чем нативный, или тем более отдельный селекты.
<details><summary>Скриншот</summary>

![изображение](https://github.com/user-attachments/assets/6c12029d-4c0b-4643-94bb-218f3a48e0e7)

</details> 

Скорее всего я буду использовать именно его в дальнейшем и хотелось бы быть в  курсе возможных подводных камней.
